### ISSUE TITLE
Invalid transformation in SLPVectorizer: Target is more poisonous than source

### ISSUE THREAD

**Issue by fhahn** [llvm/llvm-project#58391]
`SLPVectorizer` transforms the following input `@src` into `@tgt`. According to Alive2, the target is more poisonous than source.

```ll
define <4 x double> @src(i32 %a2, i32 %a3) #0 {
  %cvt2 = sitofp i32 %a2 to double
  %cvt3 = sitofp i32 %a3 to double
  %res2 = insertelement <4 x double> undef, double %cvt2, i32 0
  %res3 = insertelement <4 x double> %res2, double %cvt3, i32 1
  ret <4 x double> %res3
}

define <4 x double> @tgt(i32 %a2, i32 %a3) #0 {
  %1 = insertelement <2 x i32> poison, i32 %a2, i32 0
  %2 = insertelement <2 x i32> %1, i32 %a3, i32 1
  %3 = sitofp <2 x i32> %2 to <2 x double>
  %4 = shufflevector <2 x double> %3, <2 x double> poison, <4 x i32> <i32 0, i32 1, i32 undef, i32 undef>
  ret <4 x double> %4
}
```

https://alive2.llvm.org/ce/z/ZCaBMX

```console
Transformation doesn't verify!

ERROR: Target is more poisonous than source

Example:
i32 %a2 = poison
i32 %a3 = poison

Source:
double %cvt2 = poison
double %cvt3 = poison
<4 x double> %res2 = < poison, NaN	[based on undef value], NaN, NaN >
<4 x double> %res3 = < poison, poison, NaN, NaN >

Target:
<2 x i32> %1 = < poison, poison >
<2 x i32> %2 = < poison, poison >
<2 x double> %3 = < poison, poison >
<4 x double> %4 = < poison, poison, poison, poison >
Source value: < poison, poison, NaN, NaN >
Target value: < poison, poison, poison, poison >
```

**Comment by fhahn** [comment]
From `@sitofp_with_undef_4xi32_4f64` in `llvm/test/Transforms/SLPVectorizer/X86/sitofp.ll`

**Comment by alexey-bataev** [comment]
Fixed in 48bc5b0a297e51a0556b6e9b77c772d85c39be01
